pkg/plugin/manager.go

```diff
@@ -56,24 +52,72 @@ func pluginDir() string {
 	return pluginDir
 }
 
-func getPlugin(binary, pluginType, name string, logger hclog.Logger) (interface{}, error) {
-	pluginInfo, found := pluginBinaries[binary]
-	if !found {
-		return nil, errors.Errorf("plugin binary %q not found", binary)
+func addPlugin(plugins map[string]plugin.Plugin, kind PluginKind) map[string]plugin.Plugin {
+	var plugin plugin.Plugin
+
+	switch kind {
+	case PluginKindObjectStore:
+		plugin = &ObjectStorePlugin{}
+	case PluginKindBlockStore:
+		plugin = &BlockStorePlugin{}
 	}
 
-	cfg := pluginInfo.config
-	cfg.Cmd = exec.Command(filepath.Join(pluginInfo.dir, binary), "plugin", pluginType, name)
-	cfg.Logger = logger
+	plugins[string(kind)] = plugin
+
+	return plugins
+}
 
-	goPluginClient := plugin.NewClient(&cfg)
+func (m *manager) getPlugin(descriptor pluginInfo, logger hclog.Logger) (interface{}, error) {
+	client, found := m.clients[descriptor]
+	if !found {
+		cfg := &plugin.ClientConfig{
+			HandshakeConfig:  Handshake,
+			AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
+			Plugins:          make(map[string]plugin.Plugin),
+			Logger:           logger,
+		}
+
+		if (descriptor.kind == PluginKindBlockStore || descriptor.kind == PluginKindObjectStore) &&
+			(descriptor.name == "aws" || descriptor.name == "gcp" || descriptor.name == "azure") {
```

Comment:
Packer lets you override the built-ins. WDYT about supporting that?

Reply:
so look for an external for the kind/name, if it exists use that, else use the built-in? i like that